ui/lib/CreateStream.tsx

```diff
@@ -71,60 +117,196 @@ const CreateStream = (props: CreateStreamProps) => {
+    if (gasPrice / 1000 >= nativeBalance) {
+      return (
+        <div>
+          <h3>Not enough native tokens to pay for transaction fee</h3>
+          <button
+            onClick={() => window.open("https://hop.exchange/", "_blank")}
+          >
+            Go to Hop Exchange
+          </button>
+        </div>
+      );
+    }
+
+    // 3: Check if the transaction amount and token balance are still loading
+    if (transactionAmount === null || tokenBalance === null) {
+      return (
+        <div>
+          <p>Checking stream token balance...</p>
+        </div>
+      );
+    }
+
+    // 4: Check for enough of the selected token for the stream
+    if (transactionAmount >= tokenBalance) {
+      return (
+        <div>
+          <h3>Not enough tokens to pay for stream</h3>
+          <button
+            onClick={() => window.open("https://app.uniswap.org/", "_blank")}
+          >
+            Go to Uniswap
+          </button>
+        </div>
+      );
+    }
+
+    // 5: If all checks pass, show the slider for stream length
+    return (
+      <div>
+        <h3>Select Stream Length & approve transaction</h3>
+        <div id="CreateStream-lifetime">
+          <Slider
+            min={1}
+            marks={marks}
+            max={maxTimeDays}
+            step={null}
+            value={Math.min(selectedTime / SECS_PER_DAY, maxTimeDays)}
+            defaultValue={1}
+            onChange={(value) =>
+              typeof value === "number" && setSelectedTime(SECS_PER_DAY * value)
+            }
+          />
+        </div>
+        <br />
+        <div id="CreateStream-approve">
+          <button
+            onClick={approveStream}
+            disabled={isSuccess}
+            style={{ backgroundColor: isSuccess ? "grey" : "initial" }}
+          >
+            {`Approve ${transactionAmount} ${tokenData?.symbol}`}
+          </button>
+          {isLoading && (
+            <div>
+              Waiting for transaction confirmation to proceed to next step
+            </div>
+          )}
+          {isError && <div>Transaction Error.</div>}
+          {isSuccess && (
+            <div>
+              <h3>
+                Transaction has been confirmed. You can proceed to deployment.
+              </h3>
+              <button onClick={validateStep2}>Next</button>
+            </div>
+          )}
+        </div>
       </div>
-      <br />
+    );
+  };
 
-      {/* Part 3: The user creates their stream, using the given reason/product code */}
+  const Step3 = () => {
+    return (
       <div id="CreateStream-create">
+        <h3>Review params & Open Stream</h3>
+        <div>
+          <p>
+            <strong>Stream Manager Address:</strong>{" "}
+            {props.streamManagerAddress}
+          </p>
+          <p>
+            <strong>Token Address:</strong> {props.tokenAddress}
+          </p>
+          <p>
+            <strong>Amount Per Second:</strong> {props.amountPerSecond}
```

Comment:
Would be nicer to display total approved divided by the amount per second to show how many days you paid for